tools/sylph/sylph_profile.xml

```diff
@@ -0,0 +1,210 @@
+<tool id="sylph_profile" name="sylph profile" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
+    <expand macro='description'/>
+    <macros>
+       <import>macros.xml</import> 
+    </macros>
+    <expand macro='requirements'/>
+    <command detect_errors='exit_code'><![CDATA[
+#import re
+##SYMLINK SYLPH DB
+@DB_SELECTOR@
+##Single input
+        #if $sketch.type == 'single':
+            @SINGLE_INPUT@
+##Single group
+        #else if $sketch.type == 'single_group':
+            @SINGLE_GROUP@
+##Paired input
+        #else if $sketch.type == 'paired':
+            @PAIRED@
+##Paired  group
+        #else if $sketch.type == 'paired_group':
+            @PAIRED_GROUP@
+        #end if
+##SKETCHING
+        sylph sketch
+        #if $sketch.type == 'single':
+            $input
+        #else if $sketch.type == 'single_group':
+            -r $input
+        #else if $sketch.type == 'paired': 
+            -1 $read1
+            -2 $read2
+        #else if $sketch.type == 'paired_group': 
+            -1 $read1
+            -2 $read2
+        #end if
+        -t \${GALAXY_SLOTS:-4}
+        -d sylph_sketches &&
+##MAIN COMMAND
+        sylph profile
+        database.syldb
+        sylph_sketches/*.sylsp
+        #if $min_num_kmers:
+            --min-number-kmers ${min_num_kmers}
+        #end if
+        -t \${GALAXY_SLOTS:-4}
+        -o $output
+
+        #if $outputs:
+            #if $database_select.select == 'cached':
+                && ln -s '$database_select.sylph_database.fields.path/database.tsv.gz' 'database.tsv.gz'
+            #else:
+                && ln -s '$database_select.metadata' 'database.tsv.gz'
```

Comment:
what is `database_select.metadata` ?

Is this piece tested?

Reply:
Because of the way sylph works, it has two separate bits of data it works with -- the database itself, and the metadatafile associated with it. The metadata file MUST be the same one associated with the database, otherwise it will not work. Because of that, we set up the database file to be a directory endpoint with the intent that that directory would hold the db and the associated metadata file. In this case, if a user wants to select it from the history, we added a secondary param to the history part that lets a user optionally add that metadata file, should they want to run that part of the tool